functions/go/set-labels/transformer/labels.go

```diff
@@ -99,78 +75,365 @@ func (p *LabelTransformer) Transform(objects fn.KubeObjects) error {
 		return nil
 	}
 	for _, o := range objects.WhereNot(func(o *fn.KubeObject) bool { return o.IsLocalConfig() }) {
-		// this label need to set for all GVK
-		metaLabelsPath := FieldPath{"metadata", "labels"}
-		updatedLabels, err := updateLabels(&o.SubObject, metaLabelsPath, p.NewLabels, true)
+		err := func() error {
+			// set meta labels
+			if err := p.setObjectMeta(o); err != nil {
+				return err
+			}
+			// set selector
+			if err := p.setSelector(o); err != nil {
+				return err
+			}
+			// set PodTemplateSpec
```

Comment:
nit: I think these comments (L79, L83, and L87) are not necessary, the name of the methods are descriptive enough